src/metabase/models/collection.clj

```diff
@@ -569,27 +580,26 @@
         ;; the `FROM` clause is where we limit the collections to the ones we have permissions on. For a superuser,
         ;; that's all of them. For regular users, it's a) the collections they have permission in the DB for, and b)
         ;; their personal collection and its descendants.
-        :from [[{:union-all (if api/*is-superuser?*
-                              [{:select [:c.*]
-                                :from [[:collection :c]]}]
-                              [{:select [:c.*]
-                                :from [[:collection :c]]
-                                :join [[:permissions :p]
-                                       [:= :c.id :p.collection_id]
-                                       [:permissions_group :pg] [:= :pg.id :p.group_id]
-                                       [:permissions_group_membership :pgm] [:= :pgm.group_id :pg.id]]
-                                :where [:and
-                                        [:= :pgm.user_id api/*current-user-id*]
-                                        [:= :p.perm_type "perms/collection-access"]
-                                        [:or
-                                         [:= :p.perm_value "read-and-write"]
-                                         (when (= :read (:permission-level visibility-config))
-                                           [:= :p.perm_value "read"])]]}
-                               (when-let [personal-collection-and-descendant-ids (user->personal-collection-and-descendant-ids api/*current-user-id*)]
+        :from [(if is-superuser?
+                 [:collection :c]
+                 [{:union-all [{:select [:c.*]
+                                :from   [[:collection :c]]
+                                :join   [[:permissions :p]
+                                         [:= :c.id :p.collection_id]
+                                         [:permissions_group :pg] [:= :pg.id :p.group_id]
+                                         [:permissions_group_membership :pgm] [:= :pgm.group_id :pg.id]]
+                                :where  [:and
+                                         [:= :pgm.user_id api/*current-user-id*]
```

Comment:
what's the point of having `current-user-id` as a param to this function if we still use `api/*current-user-id*`? seems like a bug, right?

CC @crisptrutski @johnswanson

Reply:
Oops, yeah missed a spot. Thanks!